Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+                             direction=Direction.Input,
+                             doc="Sample geometry for self-attenuation correction to be applied.")
+
+        self.setPropertySettings('SampleGeometry', EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty(PropertyManagerProperty('ExperimentPropertiesDictionary', dict()),
+                             doc="Dictionary for the geometry used for self-attenuation correction.")
+
+        self.setPropertySettings('ExperimentPropertiesDictionary',
+                                 EnabledWhenProperty('SampleGeometry', PropertyCriterion.IsNotEqualTo, 'Custom'))
+
+        self.declareProperty(FileProperty(name="SampleGeometryFile",
+                             defaultValue="",
+                             action=FileAction.OptionalLoad),
+                             doc="The path to the custom geometry for self-attenuation correction to be applied.")
+
+        self.setPropertySettings('SampleGeometryFile', EnabledWhenProperty('SampleGeometry', PropertyCriterion.IsEqualTo, 'Custom'))
+
+        self.declareProperty(name="DetectorEfficiencyCalibration",
+                             defaultValue="None",
+                             validator=StringListValidator(["None", "Vanadium", "Incoherent",  "Paramagnetic"]),
+                             direction=Direction.Input,
+                             doc="Detector efficiency calibration type.")
+
+        self.setPropertySettings('DetectorEfficiencyCalibration', sample)
+
+        self.declareProperty(name="IncoherentCrossSection",
+                             defaultValue=0,
+                             validator=IntBoundedValidator(lower=0),
+                             direction=Direction.Input,
+                             doc="Cross-section for incoherent scattering, necessary for setting the output on the absolute scale.")
+
+        incoherent = EnabledWhenProperty('DetectorEfficiencyCalibration', PropertyCriterion.IsEqualTo, 'Incoherent')
+
+        absoluteNormalisation = EnabledWhenProperty('AbsoluteUnitsNormalisation', PropertyCriterion.IsDefault)
+
+        self.setPropertySettings('IncoherentCrossSection', EnabledWhenProperty(incoherent, absoluteNormalisation,
+                                                                               LogicOperator.And))
+
+        self.declareProperty(name="OutputUnits",
+                             defaultValue="Wavelength/TOF",
+                             validator=StringListValidator(["Wavelength/TOF", "TwoTheta", "Q"]),
+                             direction=Direction.Input,
+                             doc="The choice to display the reduced data either as a function of the raw data units, the detector twoTheta,"
+                                 +" or the momentum exchange.")
+
+        self.setPropertySettings('OutputUnits', EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty(name="TOFUnits",
```

Comment:
This is an option in the loader, but here it should not be, the reduction should use the option that is needed and output what is relevant based on measurement technique.

Reply:
The protocols for D7 reduction foresee using either Time-Of-Flight, time channel number, or energy as the unit for the output of TOF data, thus this choice has been exposed to the user. Do you think it should be changed?